tests/unit/models/test_sbom.py

```diff
@@ -533,22 +541,27 @@ def test_cyclonedx_sbom_can_be_converted_to_spdx_and_back_without_loosing_any_da
             ]
         )
         cyclonedx_sbom = sbom.to_spdx("NOASSERTION").to_cyclonedx()
+
+        # We have to re-order components because 'hermeto' now comes alphabetically later
+        # TODO: Stop storing components/properties as a list and use sets; that, however, will
```

Comment:
We can keep them as lists, but provide a custom `__eq__`.

Reply:
Hmm, that would work. I guess it might also turn out to be less messy than conversions, but one must not forget to update the custom `__eq__` if a new field is added/recorded in the representation, otherwise it may end up being a bigger mess in debugging than array->dict conversions.
I'll leave the commentary as is for now, it's harmless, the TODO note on its own is important, the rest is implementation detail and we can do as you suggest :).

Reply:
Sure, let's keep it and figure out the best way once it is addressed.